src/main/fc/core.c

```diff
@@ -571,10 +571,12 @@ void tryArm(void)
                 // otherwise consider only the switch position
                 crashFlipModeActive = (tryingToArm == ARMING_DELAYED_CRASHFLIP) ? false : IS_RC_MODE_ACTIVE(BOXCRASHFLIP);
 #ifdef USE_DSHOT
-                // previous disarm will have set direction to normal
-                // at this point we only need to reverse the motors if crashflipMode is active
+                // previous disarm should have set direction to normal
+                // at this point should only need to reverse the motors if crashflipMode is active, but there are bug reports.
                 if (crashFlipModeActive) {
                     setMotorSpinDirection(DSHOT_CMD_SPIN_DIRECTION_REVERSED);
+                } else {
+                    setMotorSpinDirection(DSHOT_CMD_SPIN_DIRECTION_NORMAL); //mitigate bug reports.
```

Comment:
From comments, it seems possible that `// flip was successful, continue into normal flight without need to disarm/rearm` branch may be affected too

Maybe move it out of `if (crashFlipModeActive) {` and always set direction on arm.
Crashflip logic is terrible to understand and spread over multiple places. Some larger refactoring would be great/

Reply:
open to suggestions, i'm not too confident with firmware mods, and 100% not confident to refactor myself.

Reply:
please PR, closing this one, TYVM

Reply:
Done in #14362